### PR TITLE
Vesuvio - LoadVesuvio spectrumlist parsing

### LINKED ISSUE #16097: Vesuvio - LoadVesuvio SpectrumList validation bug
### Expected behavior
* LoadVesuvio should be able to handle a comma separated list as input for SpectrumList

### Actual behavior
* Providing a comma separated list produces an error in validation

### Steps to reproduce the behavior
* Run LoadVesuvio with any data and a spectrumlist of `30-40,50-60,70-80`

### Platforms affected
* All

*I'm fairly sure the code handles this properly underneath and it is just a problem with the SpectrumList input not being separated correctly.*

### PR DESCRIPTION
**Pull request by ElliotAOram**
The Spectrum List in LoadVesuvio should now be able to handle strings in the form `10-20,30-40` or `10-20;30-40` as well as more simple cases. Tests have also been added to support these changes

**To test:**
- Ensure that you have enabled searching the data archive and try the following script checking that all the spectra numbers are correct 

```
evs_ws_comma_ranges = LoadVesuvio(Filename='20000', SpectrumList='10-20,30-40')       # expected 22 spectra
evs_ws_semi_colon_ranges = LoadVesuvio(Filename='20000', SpectrumList='10-20;30-40')  # expected 22 spectra
evs_ws_comma = LoadVesuvio(Filename='20000', SpectrumList='10,20,30,40')              # expected 4  spectra
evs_ws_range = LoadVesuvio(Filename='20000', SpectrumList='10-20')                    # expected 11 spectra
```

Fixes #16097 

[RELEASE NOTES](https://github.com/mantidproject/mantid/blob/8134bbf9f8e2324a3375e1fc8df19852d90e2f1a/docs/source/release/v3.7.0/indirect_inelastic.rst#bugfixes)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
The code changes are small and well isolated. It is good to see additional tests to verify this doesn't break in the future.

**Comment by martyngigg** [comment]
@ElliotAOram The tests are nice and the loading works without error.

I just want to check that I understand correctly. I'm assuming the ranges are inclusive, i.e `10-20` means include `20` in the loading? This would mean that I would expect that to give `11` spectra not 10 as in the instructions. Is that correct?

**Comment by ElliotAOram** [comment]
@martyngigg yes sorry you're right! it will include both 10 and 20 so it would be 11. I'll update the expected number of spectra in the PR description for clarity

**Comment by martyngigg** [comment]
Thanks!

File: Testing/SystemTests/tests/analysis/LoadVesuvioTest.py [hunk 1]
```diff
@@ -20,7 +20,34 @@ def tearDown(self):
...
+    def test_spectrum_list_comma_separated_list(self):
+        diff_mode = "DoubleDifference"
+        self._run_load("14188", "10,20,30,40", diff_mode)
+
+        # check workspace created
+        self.assertTrue(mtd.doesExist(self.ws_name))
+
+    def test_spectrum_list_comma_separated_ranges(self):
+        diff_mode = "DoubleDifference"
+        self._run_load("14188", "10-20;30-40", diff_mode, do_size_check=False)
```
Comment (martyngigg) [inline Testing/SystemTests/tests/analysis/LoadVesuvioTest.py]: What's the reason we don't check the sizes here?

**Comment by martyngigg** [comment]
@ElliotAOram The system test failure is unrelated to this. I've run the test locally on the merged branch and its fine.

File: Testing/SystemTests/tests/analysis/LoadVesuvioTest.py (hunk 1 above)
Comment (martyngigg) [inline Testing/SystemTests/tests/analysis/LoadVesuvioTest.py]: @ElliotAOram As part of the maintenance of the vesuvio tests can you look at adding the size check here too?

**Comment by martyngigg** [comment]
The system test failure is unrelated. It passes locally and the changes are sensible.

Let's :shipit:

**Comment by FedeMPouzols** [comment]
The failure in the RHEL7 + system tests is totally unrelated to this (filenames and cmake versions issue with the "viridis" color map file).

**Comment by FedeMPouzols** [comment]
Do we / should we have common parsing functions for this type of spectrum lists? I'm not sure if we have any functions that do the "Mantid SpectrumList parsing" and if they're exposed to Python. In Python algorithms that I know at least there are several Engg algorithms that do a similar parsing. And I can imagine the same or very similar parsing is done or should be done in many other places.

**Comment by martyngigg** [comment]
@FedeMPouzols You're right, we should have that. There is code in Kernel somewhere I think but it's not exposed to Python

**Comment by ElliotAOram** [comment]
@martyngigg I'm happy to export it to python if you can point me in the code for it in Kernel?

**Pull request merged**